app/src/androidTest/java/ch/epfl/sweng/swenggolf/FakeFirebaseStorage.java

```diff
@@ -52,9 +52,13 @@ private static void setUpImageWrite(final boolean working, StorageReference root
         final Task<Uri> taskUri = Mockito.mock(Task.class);
         Answer answerWrite = new Answer() {
             @Override
-            public Object answer(InvocationOnMock invocation) {
+            public Object answer(InvocationOnMock invocation) throws Exception {
                 OnCompleteListener<Uri> listener = invocation.getArgument(0);
-                listener.onComplete(taskUri);
+                if (working) {
+                    listener.onComplete(taskUri);
+                } else {
+                    throw new Exception("Storage is not working");
```

Comment:
Here i don't think the listener should do this when the storage doesn't work. Maybe check with @robinmamie who is more familiar with the code.

Reply:
I think a simple try/catch will do the job in CreateOfferActivity, right where we use `writeFile` (~line 134). I just noticed that an original exception thrown in writeFile isn't catched either, so that would solve both problems at once. We just have to not forget to create an offer with an empty link to the image, if it fails.

Also, please think of throwing an exception right where the `TODO` is, so that the tests and the real application are symmetrical. Thank you, that's my bad, I didn't handle this situation yet.